### DISCUSSION TITLE
请问一下单测生产

### DISCUSSION THREAD

**Discussion by David-Liu5460** [ant-design/ant-design#51206] [Q&A] [answered]
单测的生成是硬撸的吗？还是有什么自动化生成工具，关于单测的生成有没有什么文档可以参考呀？

**Comment by petercat-assistant** [comment]
关于 Ant Design 的单元测试生成，目前并没有提到特定的自动化生成工具。Ant Design 团队对于单元测试的态度非常谨慎，通常需要考虑组件在不同版本的 React 上的运行情况，并且在主分支更新时，还需要考虑项目构建产物在不同 React 版本上的运行情况。Ant Design 的所有组件共有 4000 多个测试用例，为了提高测试效率，团队搭建了分布式测试环境，并利用 GitHub Action 的 Job 矩阵策略来执行测试任务。

如果你需要参考更多关于单元测试的文档，可以查看 Ant Design 的 GitHub 仓库中的相关工作流配置文件，例如 [Normal test](https://github.com/ant-design/ant-design/blob/40fb753349c4f2be314c91dbb7e6f1a960097c19/.github/workflows/test.yml#L141-L223) 和 [Module test](https://github.com/ant-design/ant-design/blob/40fb753349c4f2be314c91dbb7e6f1a960097c19/.github/workflows/test.yml#L294-L357)。

如果你有更多关于单测生成的具体问题，欢迎在评论中描述你的问题并 @petercat-assistant 以便与我进行对话。

**Comment by Wxh16144** [comment, marked as answer]
你可以过一遍 jest 和 test-library/react 这些官方文档，然后留意一下 antd 一些人的 PR，这种 unit test 一般没啥难度。

**Reply by David-Liu5460** [reply]
好的，非常感谢～～～

**Comment by crazyair** [comment]
test 都是长年累月一点一点加的。

**Reply by David-Liu5460** [reply]
好的，非常感谢回复
